source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -100,14 +95,11 @@ void Cluster::addOrUpdateWorker(
 
   ENVOY_LOG(debug, "adding new dfproxy cluster host '{}'", host);
 
-  if (new_host_map == nullptr) {
-    new_host_map = std::make_shared<HostInfoMap>(*current_map);
-  }
-  const auto emplaced =
-      new_host_map->try_emplace(host, host_info,
-                                std::make_shared<Upstream::LogicalHost>(
-                                    info(), host, host_info->address(), dummy_locality_lb_endpoint_,
-                                    dummy_lb_endpoint_, nullptr, time_source_));
+  const auto emplaced = host_map_.try_emplace(
+      host, host_info,
+      std::make_shared<Upstream::LogicalHost>(info(), std::string{host}, host_info->address(),
+                                              dummy_locality_lb_endpoint_, dummy_lb_endpoint_,
+                                              nullptr, time_source_));
```

Comment:
nit: you can drop the write lock here I think.

Reply:
Fixed in 3e16973